src/snowflake/connector/network.py

```diff
@@ -453,8 +454,8 @@ def request(
         body=None,
         method: str = "post",
         client: str = "sfsql",
+        timeout: int | None = None,
         _no_results: bool = False,
-        timeout=None,
```

Comment:
if we want to make the change, we need to make sure all the callers switch the order as well